returnn/__main__.py

```diff
@@ -231,6 +231,11 @@ def load_data(config, cache_byte_size, files_config_key, **kwargs):
         kwargs["cache_byte_size"] = cache_byte_size
     Dataset.kwargs_update_from_config(config, kwargs)
     data = init_dataset(kwargs)
+  elif config.is_typed(files_config_key) and isinstance(config.typed_value(files_config_key), Dataset):
+    # Config contains dataset object, load it
+    data = config.typed_value(files_config_key)
+    data.initialize()
```

Comment:
Why do you need this here? I thought you would have a callable here.
So:
```suggestion
  elif config.is_typed(files_config_key) and callable(config.typed_value(files_config_key)):
    data = init_dataset(config.typed_value(files_config_key))
```

Reply:
Also, I would very much recommend to not directly have a `Dataset` instance inside your config. This could lead to strange problems.